WooCommerce/Classes/POS/Presentation/Barcode Scanning/GameControllerBarcodeObserver.swift

```diff
@@ -142,4 +143,100 @@ final class GameControllerBarcodeObserver {
             )
         }
     }
+
+    // MARK: - VoiceOver Support
```

Comment:
What do you think about putting much of this code in a different class? I think it's a little confusing, even with the comments, to have non-GameController approaches mixed into, and using, classes with GameController in the name.

I think we would want to keep the `GameControllerBarcodeObserver` unaware of this fallback, and switch between them from the `BarcodeScannerContainer` instead.

Apologies if I'm missing any reason that this is particularly tricky to do...

Reply:
I think that's the right idea. I chose it for simplicity since `GameControllerBarcodeObserver` has some code that I would like to reuse between the two but maybe I could move it to the container 🤔 I'll see what I can do.

Reply:
- I split the logic between `GameControllerBarcodeObserver` and `UIKitBarcodeObserver`
- Added VoiceOver switch in the `BarcodeScannerContainer` and unit tested it
- Moved shared tracking logic into a `BarcodeAnalyticsTracker`